### PR TITLE
[v12] Update device trust docs

### PR DESCRIPTION
**Pull request by flyinghermit**
Backports #30821 to branch/v12

Manually backported and contents cherrypicked as `v12` is far behind in Device Trust features. 

Careful review will be appreciated.

### CODE REVIEW

File: docs/pages/access-controls/device-trust.mdx [hunk 1]
```diff
@@ -1,31 +1,94 @@
...
 <Admonition type="warning">
-Device Trust is currently in Preview mode.
+  Device Trust is currently in Preview mode and supports following components:
 
-Device Trust requires Teleport Enterprise v12+.
+  - User devices: macOS.
+  - Teleport client: `tsh` and Teleport connect.
+  - Resources: SSH, Database and Kubernetes.
+
+  Support for Windows device is available in version 13.0 and above.
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust.mdx:15]:
```suggestion
  Support for Windows devices with a TPM is available in version 13.0 and above.
```

File: docs/pages/access-controls/device-trust.mdx [hunk 2]
```diff
@@ -1,31 +1,94 @@
...
 trusted device, in addition to establishing the user's identity and enforcing
 the necessary roles. Furthermore, users using a trusted device leave audit
 trails that include the device's information.
 
-The device trust preview works only with macOS devices and supports the
-following Teleport features:
+Device Trust requires two of the following steps to have been configured:
+
+- Trusted device registered and enrolled with Teleport.
+- Device enforcement mode configured via either a role or a cluster-wide config.
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust.mdx:32]: In v12, we only support cluster-wide config, right? I'm pretty sure role enforcement came in v13.

File: docs/pages/access-controls/device-trust.mdx [hunk 3]
```diff
@@ -1,31 +1,94 @@
...
+
+Device management is divided into two separate phases: inventory management and
+device enrollment.
+
+**Inventory management** is performed by a device admin. In this step, devices
+are registered or removed from Teleport. For example, this happens when the IT
+department of your company acquires new devices, or retires devices from use.
+
+Inventory management can be manually operated using `tctl` or automatically synced
+with a Mobile Device Management (MDM) solution such as Jamf Pro.
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust.mdx:47]: Jamf integration doesn't exist in v12 either. I would remove this.

File: docs/pages/access-controls/device-trust.mdx [hunk 4]
```diff
@@ -1,31 +1,94 @@
...
+
+Enforcing Device Trust means configuring Teleport with device trust mode, i.e. applying
+`device_trust_mode: required` rule, which tells Teleport Auth Service to only allow access
+with a trusted and an authenticated device, in addition to establishing the user's identity and enforcing
+the necessary roles.
 
-- SSH access enforcement
-- Database access enforcement
-- Kubernetes access enforcement
+Teleport supports two methods for device enforcement: Role-based
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust.mdx:84]: Not in v12.

File: docs/pages/access-controls/device-trust/device-management.mdx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+(!docs/pages/includes/device-trust/prereqs.mdx!)
+
+## Register a trusted device
+
+The `tctl` tool is used to manage the device inventory. A device admin is
+responsible for managing devices, adding new devices to the inventory and
+removing devices that are no longer in use.
+
+<Admonition type="tip" title="Self enrollment: v13.3.5+">
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/device-management.mdx:23]: Since these are v12 docs, should we just remove anything that's specific to v13 and up?

File: docs/pages/access-controls/device-trust/device-management.mdx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+```code
+$ tctl devices enroll --asset-tag="(=devicetrust.asset_tag=)"
+Run the command below on device "(=devicetrust.asset_tag=)" to enroll it:
+tsh device enroll --token=(=devicetrust.enroll_token=)
+```
+
+## Enroll a trusted device
+
+To perform the enrollment ceremony, using the device specified above, type the
+command printed by `tctl devices enroll`:
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/device-management.mdx:76]: Does `tctl devices enroll` exist in v12?

File: docs/pages/access-controls/device-trust/device-management.mdx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  Teleport version 13.3.6 and above has the preset `device-admin` role, which
+  is a substitute for the role described below.
+</Admonition>
+
+Following is an example of a role that grants permissions for the `device` resource is necessary to manage
+the inventory. Save the yaml below as `device-admin.yaml` and create it in your
+cluster:
+
+```yaml
+version: v6
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/device-management.mdx:138]: Does role v6 exist in v12 or is that a v13 addition?

File: docs/pages/includes/device-trust/prereqs.mdx [hunk 8]
```diff
@@ -0,0 +1,4 @@
+- To enroll a macOS device, you need:
+  - A signed and notarized `tsh` binary for enrollment.
+    [Download the macOS tsh installer](../../installation.mdx#macos).
+  - A Teleport version newer than v12.0.0.
```
Comment (zmb3) [inline docs/pages/includes/device-trust/prereqs.mdx:4]: "newer than v12.0.0" suggests that 12.0.0 will not work, and you need at least 12.0.1. Is that true?

**Review by zmb3 — COMMENTED** [review]
I think there's a lot of stuff in here that doesn't apply to v12 and should probably be removed.

File: docs/pages/access-controls/device-trust.mdx (hunk 2 above)
Comment (sshahcodes) [reply docs/pages/access-controls/device-trust.mdx:32]: v12 supports both cluster-wide and RBAC. v12 doc link for reference - https://goteleport.com/docs/ver/12.x/access-controls/device-trust/guide/#optional-role-based-trusted-device-enforcement

File: docs/pages/access-controls/device-trust/device-management.mdx (hunk 5 above)
Comment (sshahcodes) [reply docs/pages/access-controls/device-trust/device-management.mdx:23]: I usually add tip to tease features in new version so that reader may be tempted to upgrade. Happy to remove if it gives wrong impression though.

File: docs/pages/access-controls/device-trust/device-management.mdx (hunk 6 above)
Comment (sshahcodes) [reply docs/pages/access-controls/device-trust/device-management.mdx:76]: It does. It creates enrollment token: 
```
$ tctl devices enroll --asset-tag="C00AA0AAAA0A"
Run the command below on device "C00AA0AAAA0A" to enroll it:
tsh device enroll --token=AAAAAAAAAAAAAAAAAAAAAAAA-this-is-an-example
```

File: docs/pages/access-controls/device-trust/device-management.mdx (hunk 7 above)
Comment (sshahcodes) [reply docs/pages/access-controls/device-trust/device-management.mdx:138]: Hmm, I copied roles from existing v12 doc so I was pretty sure it exists. But your question here and above made me check if we actually backported wrong version to device trust v1 docs - https://goteleport.com/docs/ver/12.x/access-controls/device-trust/guide/#step-12-register-a-trusted-device. 
Another example of `v6` reference in role templates - https://goteleport.com/docs/ver/12.x/access-controls/guides/role-templates/#local-users.

But I found some docs using role `v5` as well. E.g. this role access request doc - https://goteleport.com/docs/ver/12.x/access-controls/access-requests/role-requests/#rbac-setup 

I am confused though 🤔 maybe role `v6` was introduced in v13 and later backported to v12 as well?

File: docs/pages/includes/device-trust/prereqs.mdx (hunk 8 above)
Comment (sshahcodes) [reply docs/pages/includes/device-trust/prereqs.mdx:4]: I will reword this as "A Teleport version v12.0.0 or above"

File: docs/pages/access-controls/device-trust.mdx (hunk 3 above)
Comment (sshahcodes) [reply docs/pages/access-controls/device-trust.mdx:47]: Do you think updating this as `....solution such as Jamf Pro (available from v13 and above)`, so readers know this feature is available in newer version is a good idea?

**Comment by sshahcodes** [comment]
friendly ping @codingllama

File: docs/pages/access-controls/device-trust.mdx [hunk 9]
```diff
@@ -1,31 +1,94 @@
...
+
+Device Trust leverages dedicated secure hardware in devices to store device credentials
+and perform device challenges. The specific implementation varies between types of devices.
+
+On macOS devices, Device Trust uses the Secure Enclave in order to store a
+device private key. That key is used to solve device challenges issued by the
+Teleport Auth Service, proving the identity of the trusted device.
+
+The signed attestation ensures that the Teleport Auth Service knows both the state
+of the device and that the request has come from the device.
```
Comment (codingllama) [inline docs/pages/access-controls/device-trust.mdx:70]: I think this is a leftover from TPM? We don't usually talk about attestation for macOS, as technically there is none.

File: docs/pages/access-controls/device-trust.mdx [hunk 10]
```diff
@@ -1,31 +1,94 @@
...
+
+That said, a device is as "trustworthy" as the enrollment process. If enrollment operator
+enrolls a malicious device to Teleport, establishing trust with Secure Enclave or TPM is
```
Comment (codingllama) [inline docs/pages/access-controls/device-trust.mdx:73]: Not sure we should mention TPM here either.

File: docs/pages/access-controls/device-trust.mdx (hunk 4 above)
Comment (codingllama) [reply docs/pages/access-controls/device-trust.mdx:84]: I think this landed in 12.1, with the first big device trust delivery.

**Comment by codingllama** [comment]
Up to you whether to pursue this one or not. It's a bit risky and I think v13 is a much better release for device trust, but still those docs would last about 6 months (a reasonably long time).

Please address Zac's comments as well.

**Comment by sshahcodes** [comment]
Hmm, the fact that it raises more question and confusion than I initially thought, looks like the risk outweighs the change. `v12` doc is self contained anyway, closing this PR.
